Cargo.toml

```diff
@@ -17,13 +17,16 @@ bench = false
 
 [lib]
 bench = false
+name = "nickel_lang"
+crate-type = ["cdylib", "lib"]
```

Comment:
I don't know if we want to make the main Nickel crate a dynamic C library by default. My current understanding is that it has implications (e.g. if you're trying to use Nickel from another Rust crate, see e.g. [this random issue][https://github.com/EmbarkStudios/rust-gpu/issues/367) or [this post](https://users.rust-lang.org/t/what-is-the-difference-between-dylib-and-cdylib/28847)). This is also a problem when compiling the same crate to both a normal Rust library and to WASM, see https://github.com/rust-lang/cargo/issues/4881.

One simple possibility is to make a very thin wrapper around `nickel-lang`, which is essentially just a crate re-exporting Nickel as it is but with a different library type. This is exactly what we do for the WASM REPL: see the `nickel-wasm-repl` directory. It could make sense to have a `nickel-lang-python` (or whatever better name you have) as well, that could be a `cdylib` without issues. You wouldn't need to have a `python-module` feature either.

Reply:
Addressed in 1ecfdf1c